04.ls/ls.rb

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+def main
+  files = Dir.glob('*').sort_by(&:downcase)
+  column_count = 3
+  row_count = files.count.ceildiv(column_count)
+  columns = split_into_columns(files, column_count, row_count)
+  output(columns, row_count)
+end
+
+def split_into_columns(files, column_count, row_count)
+  column_group = Array.new(column_count) { [] }
+  files.each_with_index do |file, index|
+    col_index = index / row_count
+    column_group[col_index] << file
+  end
+  column_group
+end
+
+def output(columns, row_count)
+  column_widths = columns.map { |words| words.map(&:size).max || 0 }
+  (0...row_count).each do |row_idx|
+    columns.each_with_index do |col, col_idx|
+      filename = col[row_idx]
+      print "#{filename.to_s.ljust(column_widths[col_idx])}  "
```

Comment:
```suggestion
      print "#{filename.ljust(column_widths[col_idx])}  " if filename
```

というように、filenameが存在するときだけprintするようにすれば to_s はいらないですし、 https://github.com/Kimmy-2ka/ruby-practices/pull/4#discussion_r2113005078 で答えてもらったような `|| 0` も不要になるんじゃないかなと思ったんですがどうでしょう？

Reply:
ご指摘ありがとうございます。
`column_widths` に `nil` が含まれてしまったことが問題だと考え、 `|| 0` を付けて対応していました。  
しかし、`filename` が存在する場合のみ出力する様にすれば、確かに`|| 0` も `.to_s` も不要になると考え、修正しました。
また、`if filename` としておくことで、読み手が「この `|| 0` はなぜ必要なのか？」と考えさせてしまうことも避けられるのではと感じました。